aws-analytics-pinpoint/src/main/java/com/amplifyframework/analytics/pinpoint/AWSPinpointAnalyticsPluginBehavior.kt

```diff
@@ -50,19 +57,18 @@ internal class AWSPinpointAnalyticsPluginBehavior(
         val metrics = mutableMapOf<String, Double>()
         analyticsEvent.properties.forEach { property ->
             val key = property.key
-            val analyticsProperty = property.value
-            when (analyticsProperty.value) {
+            when (val analyticsProperty = property.value) {
                 is AnalyticsStringProperty -> {
-                    attributes[key] = (analyticsProperty.value as AnalyticsStringProperty).value
+                    attributes[key] = analyticsProperty.value
                 }
                 is AnalyticsBooleanProperty -> {
-                    attributes[key] = (analyticsProperty.value as AnalyticsBooleanProperty).value.toString()
+                    attributes[key] = analyticsProperty.value.toString()
                 }
                 is AnalyticsIntegerProperty -> {
-                    metrics[key] = (analyticsProperty.value as AnalyticsIntegerProperty).value.toDouble()
+                    metrics[key] = analyticsProperty.value.toDouble()
```

Comment:
is `AnalyticsIntegerProperty` and `AnalyticsDoubleProperty` logic swapped here?  just looks weird to have `toDouble()` under `IntegerProperty`

Reply:
metric is a map of double so we have to convert it.

https://github.com/aws-amplify/amplify-android/blob/main/aws-analytics-pinpoint/src/main/java/com/amplifyframework/analytics/pinpoint/AWSPinpointAnalyticsPlugin.java#L217

Reply:
ah of course, thanks!